src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -190,7 +170,11 @@ private void writeOptionalModelBlob(StreamOutput output) throws IOException {
     public void writeTo(StreamOutput output) throws IOException {
         getModelMetadata().writeTo(output);
         writeOptionalModelBlob(output);
-        output.writeOptionalString(modelID);
+        output.writeString(modelID);
+    }
+
+    private static String getModelIDFromResponse(Map<String, Object> responseMap) {
+        return (String) responseMap.get(MODEL_ID);
```

Comment:
could ```responseMap.get(MODEL_ID)``` be null and typecasting is ok if its ```null```?

Reply:
@VijayanB could you comment on this?

Reply:
I agree, let's typecast only if it is not null.